io-engine/src/lvs/lvol_snapshot.rs

```diff
@@ -35,41 +35,51 @@ use strum::{EnumCount, IntoEnumIterator};
 use super::Error;
 use futures::future::join_all;
 
-#[async_trait(?Send)]
-trait AsyncIterator {
+pub trait AsyncIterator {
     type Item;
-    async fn next(&mut self) -> Option<SnapshotParams>;
+    fn parent(&mut self) -> Option<Self::Item>;
 }
 
 /// Iterator over Lvol Blobstore for Snapshot.
 pub struct LvolSnapshotIter {
-    inner: *mut spdk_blob,
+    inner_blob: *mut spdk_blob,
     inner_lvol: Lvol,
 }
 
 impl LvolSnapshotIter {
     pub fn new(lvol: Lvol) -> Self {
         Self {
-            inner: lvol.bs_iter_first(),
+            inner_blob: lvol.bs_iter_first(),
             inner_lvol: lvol,
         }
     }
 }
 
-#[async_trait(?Send)]
 /// Iterator implementation for LvolSnapshot.
 impl AsyncIterator for LvolSnapshotIter {
-    type Item = SnapshotParams;
-    async fn next(&mut self) -> Option<Self::Item> {
-        if self.inner.is_null() {
+    type Item = VolumeSnapshotDescriptor;
+    fn parent(&mut self) -> Option<Self::Item> {
+        if self.inner_blob.is_null() {
             None
         } else {
-            let current = self.inner;
-            match self.inner_lvol.bs_iter_next(current).await {
-                Some(next_blob) => self.inner = next_blob,
-                None => self.inner = std::ptr::null_mut(),
+            if let Some(parent_blob) =
+                unsafe { self.inner_lvol.bs_iter_parent(self.inner_blob) }
+            {
+                if let Some(uuid) = Lvol::get_blob_xattr(
+                    parent_blob,
+                    SnapshotXattrs::SnapshotUuid.name(),
+                ) {
+                    let snap_lvol = match UntypedBdev::lookup_by_uuid_str(&uuid)
+                    {
+                        Some(bdev) => Lvol::try_from(bdev).ok()?,
+                        None => return None,
+                    };
+                    self.inner_blob = parent_blob;
+                    self.inner_lvol = snap_lvol.clone();
+                    return snap_lvol.snapshot_descriptor(None);
+                }
```

Comment:
Can reduce nesting with ?
```suggestion
            let parent_blob = unsafe { self.inner_lvol.bs_iter_parent(self.inner_blob) }?;
            let uuid = Lvol::get_blob_xattr(
                    parent_blob,
                    SnapshotXattrs::SnapshotUuid.name(),
                )?;
            let snap_lvol = UntypedBdev::lookup_by_uuid_str(&uuid).map(|bdev| Lvol::try_from(bdev).ok())?;
            self.inner_blob = parent_blob;
            self.inner_lvol = snap_lvol.clone();
            snap_lvol.snapshot_descriptor(None)
```